src/store/index-level.ts

```diff
@@ -0,0 +1,205 @@
+      if (property === '_id') {
+        continue;
+      }
+
+      const value = entry[property];
+
+      const keyPrefix = this.generateKey(property, value);
+      ops.push({ type: 'put', key: keyPrefix + VALUE_DELIMITER + _id, value: _id });
+      keyPrefixes.push(keyPrefix);
+    }
+    ops.push({ type: 'put', key: `__${_id}__keyPrefixes`, value: keyPrefixes.join(KEY_PREFIX_DELIMITER) });
+
+    await abortOr(options?.signal, this.db.batch(ops));
+  }
+
+  async query(filter: Filter, options?: IndexLevelOptions): Promise<Array<string>> {
+    options?.signal?.throwIfAborted();
+
+    let requiredMatchCount = 0;
+
+    const promises: Promise<Matches>[] = [ ];
+    for (const propertyName in filter) {
+      const propertyFilter = filter[propertyName];
+
+      if (typeof propertyFilter === 'object' && propertyFilter !== null) {
+        if (Array.isArray(propertyFilter)) {
+          for (const propertyValue of propertyFilter) {
+            promises.push(this.findExactMatches(propertyName, propertyValue, options));
+          }
+        } else {
+          promises.push(this.findRangeMatches(propertyName, propertyFilter, options));
+        }
+      } else {
+        promises.push(this.findExactMatches(propertyName, propertyFilter, options));
+      }
+
+      ++requiredMatchCount;
+    }
+
+    const docIds = new Set<string>();
+    const matchCount: { [docId: string]: number } = { };
+    for (const matches of await Promise.all(promises)) {
+      for (const value of matches.values()) {
+        matchCount[value] = (matchCount[value] ?? 0) + 1;
+        if (matchCount[value] === requiredMatchCount) {
+          docIds.add(value);
+        }
+      }
+    }
+    return Array.from(docIds);
+  }
+
+  async delete(id: string, options?: IndexLevelOptions): Promise<void> {
+    options?.signal?.throwIfAborted();
+
+    const keyPrefixes = await abortOr(options?.signal, this.db.get(`__${id}__keyPrefixes`));
+    if (!keyPrefixes) {
+      return;
+    }
+
+    const ops: AbstractBatchDelOperation<LevelDatabase<string, string>, string>[] = [ ];
+    for (const keyPrefix of keyPrefixes.split(KEY_PREFIX_DELIMITER)) {
+      ops.push({ type: 'del', key: keyPrefix + VALUE_DELIMITER + id });
+    }
+    ops.push({ type: 'del', key: `__${id}__keyPrefixes` });
+
+    await abortOr(options?.signal, this.db.batch(ops));
+  }
+
+  clear(): Promise<void> {
+    return this.db.clear();
+  }
+
+  private async findExactMatches(propertyName: string, propertyValue: unknown, options?: IndexLevelOptions): Promise<Matches> {
+    const propertyKey = this.generateKey(propertyName, propertyValue);
+
+    const iteratorOptions: AbstractIteratorOptions<string, string> = {
+      gt: propertyKey
+    };
+
+    return this.findMatches(propertyKey, iteratorOptions, options);
+  }
+
+  private async findRangeMatches(propertyName: string, range: RangeFilter, options?: IndexLevelOptions): Promise<Matches> {
+    const propertyKey = this.generateKey(propertyName);
```

Comment:
it's mainly a consistency thing so that if we ever decide to adjust how `generateKey` works then it'll immediately apply everywhere instead of all places except this one